src/Refactoring-UI/RBRenameInstanceVariableDriver.class.st

```diff
@@ -0,0 +1,83 @@
+"
+I'm a model for user interaction for the rename method refactoring.
+"
+Class {
+	#name : 'RBRenameInstanceVariableDriver',
+	#superclass : 'RBInteractionDriver',
+	#instVars : [
+		'oldName',
+		'class',
+		'newName',
+		'shouldEscape',
+		'lastName'
+	],
+	#category : 'Refactoring-UI-Drivers',
+	#package : 'Refactoring-UI',
+	#tag : 'Drivers'
+}
+
+{ #category : 'execution' }
+RBRenameInstanceVariableDriver >> applyChanges [ 
+	refactoring privateTransform.
+	refactoring performChanges
+]
+
+{ #category : 'execution' }
+RBRenameInstanceVariableDriver >> configureRefactoring [
+
+	refactoring := RBRenameInstanceVariableRefactoring
+		               variable: oldName
+		               class: class
+]
+
+{ #category : 'accessing' }
+RBRenameInstanceVariableDriver >> defaultModel [
+	^ RBNamespace onEnvironment: RBBrowserEnvironment new
+]
+
+{ #category : 'execution' }
+RBRenameInstanceVariableDriver >> requestNewNameBasedOn: aName [
+
+	newName := SpRequestDialog new
+		           title: 'Please provide a new variable name';
+		           text: aName;
+		           openModal.
+
+	newName ifNil: [
+		shouldEscape := true.
+		^ self ].
+	newName = oldName ifTrue: [
+		shouldEscape := true.
+		^ self ].
+	lastName := newName.
+	^ newName
+]
+
+{ #category : 'execution' }
+RBRenameInstanceVariableDriver >> runRefactoring [
+
+	| failedConditions |
+	self configureRefactoring.
+	
+	shouldEscape := false.
+
+	[ newName := self requestNewNameBasedOn: lastName.
+	shouldEscape ifTrue: [ ^ self ].
+	refactoring newName: newName.
+	failedConditions := refactoring failedApplicabilityPreconditions.
+	failedConditions isEmpty ]
+		whileFalse: [
+			failedConditions do: [ :cond | 
+				self inform: cond errorString ] ].
```

Comment:
Currently, for `(RBNameIsGlobal ... className: name) not` inform message shows `unkonwn`. I think it's because it doesn't have `errorString` implemented. But might be because not doesn't work nicely with new violator architecture.
I'll create an issue for this and we can deal with it after this PR.

Reply:
Yes, to fix this "unknown" display , only for the moment I avoid the usage of `errorString` and instead I directly use the `violationMessageOn:` to display the error message.